x/group/types.go

```diff
@@ -118,6 +113,84 @@ func (p *ThresholdDecisionPolicy) Validate(g GroupInfo) error {
 	return nil
 }
 
+// Implements DecisionPolicy Interface
+var _ DecisionPolicy = &PercentageDecisionPolicy{}
+
+// NewPercentageDecisionPolicy creates a new percentage DecisionPolicy
+func NewPercentageDecisionPolicy(percentage string, timeout time.Duration) DecisionPolicy {
+	return &PercentageDecisionPolicy{percentage, timeout}
+}
+
+func (p PercentageDecisionPolicy) ValidateBasic() error {
+	percentage, err := math.NewPositiveDecFromString(p.Percentage)
+	if err != nil {
+		return sdkerrors.Wrap(err, "percentage threshold")
+	}
+	value, err := percentage.Float64()
+	if err != nil {
+		return sdkerrors.Wrap(err, "Dec to Float64 conversion of percentage")
+	}
+	if !(value >= 0 && value <= 1) {
+		return sdkerrors.Wrap(errors.ErrInvalid, "percentage must be in the range of 0 to 1")
+	}
+
+	timeout := p.Timeout
+	if timeout <= time.Nanosecond {
+		return sdkerrors.Wrap(errors.ErrInvalid, "timeout")
+	}
+	return nil
+}
+
+func (p *PercentageDecisionPolicy) Validate(g GroupInfo) error {
+	return nil
+}
+
+// Allow allows a proposal to pass when the tally of yes votes equals or exceeds the percentage threshold before the timeout.
+func (p PercentageDecisionPolicy) Allow(tally Tally, totalPower string, votingDuration time.Duration) (DecisionPolicyResult, error) {
+	timeout := p.Timeout
+	if timeout <= votingDuration {
+		return DecisionPolicyResult{Allow: false, Final: true}, nil
+	}
+
+	percentage, err := math.NewNonNegativeDecFromString(p.Percentage)
+	if err != nil {
+		return DecisionPolicyResult{}, err
+	}
+	yesCount, err := math.NewNonNegativeDecFromString(tally.YesCount)
+	if err != nil {
+		return DecisionPolicyResult{}, err
+	}
+	totalCounts, err := tally.TotalCounts()
+	if err != nil {
+		return DecisionPolicyResult{}, err
+	}
+
+	totalPowerDec, err := math.NewNonNegativeDecFromString(totalPower)
+	if err != nil {
+		return DecisionPolicyResult{}, err
+	}
+	undecided, err := math.SubNonNegative(totalPowerDec, totalCounts)
+	if err != nil {
+		return DecisionPolicyResult{}, err
+	}
+	sum, err := yesCount.Add(undecided)
+	if err != nil {
+		return DecisionPolicyResult{}, err
+	}
+	sumPercentage, err := sum.Quo(totalPowerDec)
+	if err != nil {
+		return DecisionPolicyResult{}, err
+	}
+
+	if sumPercentage.Cmp(percentage) >= 0 {
```

Comment:
this is wrong, `sumPercentage` represents (yes + undecided) / total power, so we can't say whether the proposal will pass or not with this value...
we need to check for yes / total power